contracts/EthereumRuntime.sol

```diff
@@ -112,7 +112,7 @@ contract EthereumRuntime is EVMConstants, IEthereumRuntime {
 
     // Init EVM with given stack and memory and execute from the given opcode
     // solhint-disable-next-line function-max-lines
-    function execute(EVMPreimage memory img) public pure returns (EVMPreimage memory) {
+    function execute(EVMPreimage memory img) public pure returns (EVMResult memory) {
```

Comment:
Why `EVMResult` is needed here?
I think most of its attributes is the same as `EVMPreimage`?
I think `stateHash(EVMPreimage)` function should be enough?

Reply:
@peara
`EVMResult` 's attributes are parameters of EVM which are passed down to stateHash so i thought instead of using the earlier `result` (whose parameters were random) we may use a result with better info.
Yes `stateHash(EVMPreimage)` will be enough but do we want the same struct to behave both as input and output of functions?

Reply:
I think using one struct could has benefits such as I can pass it to the next `execute`. but as things are hashed, we can't do that.
So most of the info is redundant actually..
I don't know for sure, but maybe just return hash could be better

Reply:
From my point of view, you can leave it as is for now. Because some things will change, that isn't going to be a problem anyway 🙂. We will probably use custom functions to invoke the EVM in different parts of the code, so leaving it as is should be fine for the tests/fixtures.